src/api/app/controllers/webui/requests_listing_controller.rb

```diff
@@ -23,25 +25,62 @@ def index
 
   # Initialize shared attributes
   def assign_attributes
-    @url = requests_path
+    @url = if @project
+             project_requests_beta_path(@project)
+           else
+             requests_path
+           end
   end
 
+  # rubocop:disable Metrics/PerceivedComplexity
   def filter_requests
-    if params[:requests_search_text].present?
-      initial_bs_requests = filter_by_text(params[:requests_search_text])
-      params[:ids] = filter_by_involvement(@filter_involvement).ids
+    initial_bs_requests = if params[:requests_search_text].present?
+                            filter_by_text(params[:requests_search_text])
+                          elsif @project
+                            BsRequest.with_actions.where('bs_request_actions.target_project = (?) OR bs_request_actions.source_project = (?)', @project.name, @project.name)
+                          else
+                            filter_by_involvement(@filter_involvement)
+                          end
+
+    if @project
+      request_ids_for_project
+      params[:project] = @project.name
```

Comment:
So if somebody uses the search box, the page stops filtering by project? Why not move `with_actions.where('bs_request_actions.target_project = (?) OR bs_request_actions.source_project = (?)', @project.name, @project.name)` to the `Query` class to avoid having to do this dance here?